assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -86,8 +94,13 @@ const usePaymentMethodRegistration = (
 				[ paymentMethod.name ]: paymentMethod,
 			};
 		};
-		for ( const paymentMethodName in registeredPaymentMethods ) {
+
+		for ( let i = 0; i < paymentMethodsOrder.length; i++ ) {
```

Comment:
Nit: you can use `forEach` if you want:
```JS
await paymentMethodsOrder.forEach( async ( paymentMethodName ) => {
  ...
} );
```

Below, you would need to change the `continue` with `return`. Not blocking, just a suggestion in case you think it would make code cleaner. :slightly_smiling_face: